hopp/simulation/technologies/financial/custom_financial_model.py

```diff
@@ -160,20 +155,44 @@ def __init__(self,
         self.batt_annual_discharge_energy = None
         self.batt_annual_charge_energy = None
         self.batt_annual_charge_from_system = None
-        self.battery_total_cost_lcos = None             # not currently used but referenced
-        self.system_use_lifetime_output = None          # Lifetime
-        self.cp_capacity_credit_percent = None          # CapacityPayments
+        self.battery_total_cost_lcos = None            # not currently used but referenced
+        self.system_use_lifetime_output = 0            # Lifetime
+        self.cp_capacity_credit_percent = [0]          # CapacityPayments
 
         # Input parameters within dataclasses
-        self.BatterySystem: BatterySystem = BatterySystem.from_dict(fin_config)
-        self.SystemCosts: SystemCosts = SystemCosts.from_dict(fin_config)
-        self.Revenue: Revenue = Revenue.from_dict(fin_config)
-        self.FinancialParameters: FinancialParameters = FinancialParameters.from_dict(fin_config)
+        if 'battery_system' in fin_config:
+            self.BatterySystem: BatterySystem = BatterySystem.from_dict(
+                fin_config['battery_system']
+            )
+        else:
+            self.BatterySystem: BatterySystem = BatterySystem()
+
+        if 'system_costs' in fin_config:
+            self.SystemCosts: SystemCosts = SystemCosts.from_dict(fin_config['system_costs'])
+        else:
+            self.SystemCosts: SystemCosts = SystemCosts()
+
+        if 'revenue' in fin_config:
+            self.Revenue: Revenue = Revenue.from_dict(fin_config['revenue'])
```

Comment:
Noting to convert this class to use attrs in the future so that we can clean up this default setting code.